### DISCUSSION TITLE
[Feature Request] Add configuration option to allow WebSocket configuration of KeepAliveInterval

### DISCUSSION THREAD

**Discussion by jamdavi** [Azure/azure-iot-sdk-csharp#2206] [Ideas]
https://stackoverflow.com/questions/69283367/how-to-set-microsoft-azure-devices-client-deviceclient-keepaliveinseconds-to-a-h

Our client implementation doesn't support a way to set the KeepAliveInterval for a WebSocketClient. This configuration option would be helpful in scenarios where a customer needs to limit the amount of traffic coming from the client.

A quick test shows that even setting the WebSocket KeepAliveInterval to 0 is acceptable and the connection will stay open due to the scheduled ping.

**Comment by drwill-ms** [comment]
This feature has been added in version 1.41.0.

<https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/2022-5-6>
